### PR TITLE
Paul/rsync builds

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204836801484055
  - https://app.asana.com/0/0/1204846049105749

### CODE REVIEW

File: scripts/deploy.ts [hunk 1]
```diff
@@ -374,6 +376,20 @@ function buildCommonPost(buildObj: BuildObj) {
...
+
+    // const rsyncDir = join(buildObj.rsyncLocation, repoBranch, platformType, String(buildNum))
+
+    const datePrefix = new Date().toISOString().slice(2, 19).replace(/:/gi, '').replace(/-/gi, '')
+    const rsyncFile = escapePath(`${datePrefix}--${productNameClean}--${platformType}--${repoBranch}--${buildNum}--${guiHash.slice(0, 8)}.ipa`)
+
+    // call(`rsync -avz -e "ssh -i ${githubSshKey}" ${buildObj.ipaFile} ${rsyncDir}/`)
+    call(`rsync -avz -e "ssh -i ${githubSshKey}" ${buildObj.ipaFile} ${(join(buildObj.rsyncLocation), rsyncFile)}`)
+    mylog('\n*** Upload to rsyncLocation Complete ***')
+  }
```
Comment (samholmes) [inline scripts/deploy.ts]: You should co-locate line 328 within his section of code (`const { buildNum, guiHash, platformType, productNameClean, repoBranch } = buildObj`). This destructuring is for line 387, so maybe put it above that line?

File: scripts/deploy.ts [hunk 2]
```diff
@@ -73,6 +77,17 @@ interface BuildObj extends BuildConfigFile {
   ipaFile: string // Also APK
 }
 
+const asLatestTestFile = asObject({
```
Comment (samholmes) [inline scripts/deploy.ts:80]: Use `asJSON` and `asCodec`?

File: scripts/deploy.ts [hunk 3]
```diff
@@ -383,16 +398,75 @@ function buildCommonPost(buildObj: BuildObj) {
...
     mylog(`\n\nUploading to rsyncLocation ${buildObj.rsyncLocation}`)
-    mylog('***********************\n')
+    mylog('***********************************************************************\n')
 
     // const rsyncDir = join(buildObj.rsyncLocation, repoBranch, platformType, String(buildNum))
 
     const datePrefix = new Date().toISOString().slice(2, 19).replace(/:/gi, '').replace(/-/gi, '')
-    const rsyncFile = escapePath(`${datePrefix}--${productNameClean}--${platformType}--${repoBranch}--${buildNum}--${guiHash.slice(0, 8)}.ipa`)
+    const fileSuffixArr = buildObj.ipaFile.split('.')
+    const fileSuffix = fileSuffixArr[fileSuffixArr.length - 1]
```
Comment (samholmes) [inline scripts/deploy.ts:407]: Propose using the word "extension" or "ext" instead of "suffix". You can also shorten this to one line.

```
    const [fileExtension] = buildObj.ipaFile.split('.').reverse()
```

File: scripts/deploy.ts (hunk 2 above)
Comment (samholmes) [reply scripts/deploy.ts:80]: Hmm, upon further review, the cleaner is never really used. Why not just a static type then?

**Review by samholmes — CHANGES_REQUESTED** [review]

File: scripts/deploy.ts (hunk 2 above)
Comment (paullinator) [reply scripts/deploy.ts:80]: It was going to be used and for now the data doesn't need to be read but only written. Will change

**Review by samholmes — APPROVED** [review]
